extensions-core/kinesis-indexing-service/pom.xml

```diff
@@ -0,0 +1,202 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.apache.druid.extensions</groupId>
+  <artifactId>druid-kinesis-indexing-service</artifactId>
+  <name>druid-kinesis-indexing-service</name>
+  <description>druid-kinesis-indexing-service</description>
+
+  <packaging>pom</packaging>
+
+  <parent>
+    <groupId>org.apache.druid</groupId>
+    <artifactId>druid</artifactId>
+    <version>0.13.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+
+  <properties>
+    <kinesis.version>1.6.3</kinesis.version>
+    <aws.version>1.10.61</aws.version>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.druid</groupId>
+      <artifactId>druid-api</artifactId>
+      <version>${project.parent.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.druid</groupId>
+      <artifactId>druid-indexing-service</artifactId>
+      <version>${project.parent.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.druid</groupId>
+      <artifactId>druid-server</artifactId>
+      <version>${project.parent.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.netty</groupId>
+      <artifactId>netty</artifactId>
+      <version>3.10.4.Final</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>com.fasterxml.jackson.dataformat</groupId>
+      <artifactId>jackson-dataformat-cbor</artifactId>
+      <version>${jackson.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>amazon-kinesis-client</artifactId>
```

Comment:
The KCL is licensed under the Amazon Software License.

I know this has had release implications for other Apache projects, I wonder if this should be considered by Druid as well?

Reply:
That's a good point. The Amazon Software License is forbidden for Apache projects based on https://www.apache.org/legal/resolved.html. In particular it has a field of use restriction, which is incompatible with Apache philosophy and license:

> 3.3 Use Limitation. The Work and any derivative works thereof only may be used or intended for use with the web services, computing platforms or applications provided by Amazon.com, Inc. or its affiliates, including Amazon Web Services, Inc.

The aws-java-sdk-kinesis library does not have any such limitation (it is Apache licensed). Is it possible to use this instead of amazon-kinesis-client? If so, we should do that. If not, I think we can still ship the kinesis extension, but we cannot include amazon-kinesis-client. It should be something users download separately and drop in the jar for on their own: https://www.apache.org/legal/resolved.html#optional

Reply:
FWIW this is the same basic situation as the mysql extension is in (the mysql driver is under a category X license) so we should probably package it the same way.

Reply:
@gianm it seems that aws-java-sdk-kinesis would work except it doesn't have `UserRecord.deaggregate`, which is needed in the KinesisRecordSupplier implementation. Do you know of any work arounds?

Reply:
assuming that the KCL was used here in the "normal" manner, ie handling leases, resharding, backoff, etc, there will be a quite large amount of work to be done to remove the KCL and still have a stable kinesis reader.

Reply:
@dyanarose ah, we don't actually use the KCL in the "normal" manner here; the library is included mainly for `IKinesisProxy` which is the Java wrapper around the HTTP API. We handle the balancing, checkpointing, and resharding in the indexing service logic (out of necessity to be able to support exactly-once read semantics)